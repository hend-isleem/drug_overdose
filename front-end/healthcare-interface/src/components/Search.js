import React, { useState } from 'react';

const Search = () => {
    const [firstMed, setFirstMed] = useState('');
    const [secondMed, setSecondMed] = useState('');
    const [interactionWarning, setInteractionWarning] = useState(null);
    const [step, setStep] = useState(1); // To control the step of input

    const handleFirstSearch = async () => {
        if (!firstMed) return;
        setStep(2); // Move to next step to enter second medication
        
    };

    const handleSecondSearch = async () => {
        if (!secondMed) return;
        
        try {
            // Simulate API call to check interactions
            const response = await fetch(`https://api.yourservice.com/interactions?drug1=${firstMed}&drug2=${secondMed}`);
            const data = await response.json();
            
            if (data.warning) {
                setInteractionWarning(data.warning);
            } else {
                setInteractionWarning("No interactions found.");
            }
        } catch (error) {
            console.error('Failed to fetch interaction information:', error);
            setInteractionWarning("Error fetching data.");
        }
    };

    return (
        <div className="search-container">
            {step === 1 ? (
                <div>
                    <input
                        type="text"
                        value={firstMed}
                        onChange={(e) => setFirstMed(e.target.value)}
                        placeholder="Enter first medication"
                    />
                    <button onClick={handleFirstSearch}>Search</button>
                </div>
            ) : (
                <div>
                    <input
                        type="text"
                        value={secondMed}
                        onChange={(e) => setSecondMed(e.target.value)}
                        placeholder="Enter second medication"
                    />
                    <button onClick={handleSecondSearch}>Check Interactions</button>
                </div>
            )}
            {interactionWarning && <div className="interaction-warnings">{interactionWarning}</div>}
        </div>
    );
};

export default Search;






