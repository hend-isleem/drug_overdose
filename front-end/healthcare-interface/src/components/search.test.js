import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('Search Component Tests', () => {
  test('fetching interactions successfully', async () => {
    global.fetch.mockResolvedValueOnce({
      json: () => Promise.resolve({ warning: 'Interactions may increase side effects.' })
    });

    const { getByPlaceholderText, getByText } = render(<Search />);
    fireEvent.change(getByPlaceholderText('Enter first medication'), { target: { value: 'Aspirin' } });
    fireEvent.click(getByText('Search'));
    fireEvent.change(getByPlaceholderText('Enter second medication'), { target: { value: 'Ibuprofen' } });
    fireEvent.click(getByText('Check Interactions'));

    await waitFor(() => {
      expect(getByText('Interactions may increase side effects.')).toBeInTheDocument();
    });
  });

  test('interaction fetch fails with network error', async () => {
    global.fetch.mockRejectedValueOnce(new Error('Network error'));

    const { getByPlaceholderText, getByText, findByText } = render(<Search />);
    fireEvent.change(getByPlaceholderText('Enter first medication'), { target: { value: 'Aspirin' } });
    fireEvent.click(getByText('Search'));
    fireEvent.change(getByPlaceholderText('Enter second medication'), { target: { value: 'Ibuprofen' } });
    fireEvent.click(getByText('Check Interactions'));

    const errorText = await findByText('Error fetching data.');
    expect(errorText).toBeInTheDocument();
  });

  // Snapshot test for Search component
  test('Search component snapshot', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});



