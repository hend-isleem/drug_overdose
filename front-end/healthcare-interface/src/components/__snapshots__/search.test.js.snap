// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Tests Search component snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-container"
  >
    <div>
      <input
        placeholder="Enter first medication"
        type="text"
        value=""
      />
      <button>
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;
